Mathlib/Data/Fin/Parity.lean

```diff
@@ -0,0 +1,42 @@
+/-
+Copyright (c) 2023 Yury Kudryashov. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yury Kudryashov
+-/
+import Mathlib.Data.Fin.Basic
+import Mathlib.Data.Nat.Parity
+
+/-!
+# Parity in `Fin n`
+
+In this file we prove that an element `k : Fin n` is even in `Fin n`
+iff `n` is odd or `Fin.val k` is even.
+-/
+
+namespace Fin
+
+/-- In `Fin n`, all elements are even for odd `n`,
+otherwise an element is even iff its `Fin.val` value is even. -/
+lemma even_iff {n : ℕ} {k : Fin n} : Even k ↔ Odd n ∨ Even k.val := by
+  constructor
+  · rintro ⟨k, rfl⟩
+    rw [or_iff_not_imp_left, ← Nat.even_iff_not_odd]
```

Comment:
It might make sense to bake this into the statement, as `Even k ↔ (Even n → Even k.val)`. I think it would shorten the proof here and below.